src/foam/u2/view/OverlayActionListView.js

```diff
@@ -216,12 +177,36 @@ foam.CLASS({
   ],
 
   listeners: [
-    function onMouseOver() {
-      this.hovered_ = true;
+    function click(evt) {
+      this.SUPER(evt);
+      this.overlay_.parentEl = this;
+      this.isMouseClick = !! evt.detail;
+      var x = evt.clientX || this.getBoundingClientRect().x;
+      var y = evt.clientY || this.getBoundingClientRect().y;
+      if ( this.disabled_ ) return;
+      if ( ! this.overlayInitialized_ ) this.initializeOverlay();
+      this.overlay_.open(x, y);
+      (this.firstEl_ && ! this.isMouseClick) && this.firstEl_.focus();
     },
 
-    function onMouseOut() {
-      this.hovered_ = false;
+    function onKeyDown(e) {
+      var isTabPressed = (e.key === 'Tab' || e.keyCode === 9);
+
+      if ( ! isTabPressed ) {
+        return;
+      }
+
+      if ( e.shiftKey ) {
+        if ( this.document.activeElement === this.document.getElementById(this.firstEl_.id) ) {
```

Comment:
Is this not the same as this.firstEl_.el()?

Reply:
It is. Didn't know we had that function. Fixed